### PR TITLE
PWGJE : Jet charm hadronization

### PR DESCRIPTION
**Pull request by christianreckziegel**
PWGJE : task created for studying the D0 and Lambda_c hadronization differences. 
That is done mainly with the jet-HF candidate distance in eta-phi plane observable.
Currently only D0 jets are being considered.
Lambda_c jets will be included in the analysis in the future.

### CODE REVIEW

File: PWGJE/Tasks/CMakeLists.txt [hunk 1]
```diff
@@ -124,4 +124,8 @@ if(FastJet_FOUND)
                         SOURCES jetLundReclustering.cxx
                         PUBLIC_LINK_LIBRARIES O2::Framework O2Physics::PWGJECore O2Physics::AnalysisCore FastJet::FastJet FastJet::Contrib
                         COMPONENT_NAME Analysis)
+    o2physics_add_dpl_workflow(jet-charm-hadronization
+                        SOURCES crCharmHadronization.cxx
+                        PUBLIC_LINK_LIBRARIES O2::Framework O2Physics::PWGJECore O2Physics::AnalysisCore
```
Comment (nzardosh) [inline PWGJE/Tasks/CMakeLists.txt:129]: please change the name of your task to not include cr at the beggining. Something like hffragmentationfunction.cxx

File: PWGJE/Tasks/crCharmHadronization.cxx [hunk 2]
```diff
@@ -0,0 +1,159 @@
...
+
+struct crCharmHadronizationTask {
+  // producing new table
+  Produces<aod::JetDistanceTable> distJetTable;
+
+  // Histogram registry: an object to hold your histograms
+  HistogramRegistry registry{"histos", {}, OutputObjHandlingPolicy::AnalysisObject};
+
+  // histogram configurables
+  Configurable<int> distBins{"distBins", 1000, "number of bins in distance histogram"};
```
Comment (nzardosh) [inline PWGJE/Tasks/crCharmHadronization.cxx:84]: does this need to be a configurable? probably fine to hard code it

File: PWGJE/Tasks/crCharmHadronization.cxx [hunk 3]
```diff
@@ -0,0 +1,159 @@
...
+      registry.fill(HIST("h_jet_counter"), 0.5);
+      // obtaining jet 3-vector
+      TVector3 jetVector(jet.px(), jet.py(), jet.pz());
+
+      for (auto& d0Candidate : jet.hfcandidates_as<CandidatesD0Data>()) { // for jet constituents use -> auto& jetConstituent : jet.tracks_as<JetTracks>()
+
+        // obtaining jet 3-vector
+        TVector3 d0Vector(d0Candidate.px(), d0Candidate.py(), d0Candidate.pz());
+        // calculating fraction of the jet momentum carried by the D0 along the direction of the jet axis
+        double z_parallel = (jetVector * d0Vector) / (jetVector * jetVector);
```
Comment (nzardosh) [inline PWGJE/Tasks/crCharmHadronization.cxx:126]: is this operand the dot product or the cross product?

Comment (christianreckziegel) [reply PWGJE/Tasks/crCharmHadronization.cxx:126]: It's the scalar dot product

File: PWGJE/Tasks/CMakeLists.txt (hunk 1 above)
Comment (christianreckziegel) [reply PWGJE/Tasks/CMakeLists.txt:129]: All right!

File: PWGJE/Tasks/crCharmHadronization.cxx (hunk 2 above)
Comment (christianreckziegel) [reply PWGJE/Tasks/crCharmHadronization.cxx:84]: All right!

**Pull request merged**